packages/plugin-react/src/index.ts

```diff
@@ -135,7 +141,13 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
           }
           // Even if the automatic JSX runtime is not used, we can still
           // inject the React import for .jsx and .tsx modules.
-          else if (!isNodeModules && !/(^|\n)import React /.test(code)) {
+          else if (
+            !isNodeModules &&
+            !(
+              importReactRegex.test(code) ||
+              importReactRegex.test(jsxInject ?? '')
```

Comment:
We should also warn the user that `jsxInject` is unnecessary when using this plugin.

Something like this:
```ts
// in `configResolved` hook
logger = config.logger

// in `transform` hook
logger.warnOnce(
  '[vite-plugin-react] This plugin imports React for you automatically, ' +
    'so you can stop using `esbuild.jsxInject` for that purpose.'
)
```